fms_extras/utils/cache/__init__.py

```diff
@@ -0,0 +1,169 @@
+import abc
+import dataclasses
+from typing import List, Optional, Tuple
+
+import torch
+
+
+@dataclasses.dataclass
+class CacheDataLayer(metaclass=abc.ABCMeta):
```

Comment:
i think we still have a single-use interface here?

Reply:
removed in the cache PR, will fix this one.